Makefile

```diff
@@ -13,79 +13,50 @@ EXAMPLE     = example
 INTDIR      = $(BUILDDIR)/obj
 LIBDIR      = $(BUILDDIR)/lib
 BINDIR      = $(BUILDDIR)/bin
-PREFIX     ?= $(DESTDIR)/usr/local
-INSTALLLIB  = $(PREFIX)/lib
-INSTALLINC  = $(PREFIX)/include
-INSTALLSH   = build-aux/install-sh
+PREFIX     ?= /usr/local
+INSTALLLIB  = $(DESTDIR)$(PREFIX)/lib
+INSTALLINC  = $(DESTDIR)$(PREFIX)/include
+INSTALLSH   = ./build-aux/install-sh
 RANLIB     ?= ranlib
 LDCONFIG   ?= ldconfig
 SHELL      := $(shell env sh -c 'PATH="$$(command -p getconf PATH)" command -v sh')
 
-# base CFLAGS
-CFLAGS += -Wall -Wextra -Wpedantic -std=c11 -I. -fPIC
-
-# MinGW compiler-specific flags
-ifneq "$(findstring mingw,$(CC))" ""
-	MINGW ?= 1
-endif
-ifeq ($(MINGW),1)
-  ifneq "$(findstring gcc,$(CC))" ""
-	CFLAGS += -Wno-unknown-pragmas
-  endif
-	MINGW_LIBS=-lshlwapi -lws2_32
-endif
-
-# Oracle compiler-specific flags
-ifneq "$(findstring suncc,$(CC))" ""
-	SUNPRO ?= 1
-endif
-ifeq ($(SUNPRO),1)
-	CFLAGS        += -fcommon
-	FORTIFY_FLAGS ?= -U_FORTIFY_SOURCE
-	MMDOPT        ?= -xMMD
-	PTHOPT        ?= -mt=yes
-else
-	FORTIFY_FLAGS ?= -D_FORTIFY_SOURCE=2
-	MMDOPT        ?= -MMD
-	PTHOPT        ?= -pthread
-endif
+# platform specifics
+include sirplatform.mk
 
-# MinGW MSVCRT workaround
-ifeq ($(SIR_MSVCRT_MINGW),1)
-	CFLAGS += -DSIR_MSVCRT_MINGW
+# base CFLAGS
+ifndef NO_DEFAULT_CFLAGS
+  CFLAGS += -Wall -Wextra -Wpedantic -std=c11 -I. -fPIC
 endif
 
 # debug/non-debug CFLAGS
 ifeq ($(SIR_DEBUG),1)
-	CFLAGS += -g -O0 -DDEBUG -U_FORTIFY_SOURCE
+  CFLAGS += -g -O0 -DDEBUG -U_FORTIFY_SOURCE
 else
-	CFLAGS += -O3 -DNDEBUG $(FORTIFY_FLAGS)
+  CFLAGS += -O3 -DNDEBUG $(FORTIFY_FLAGS)
 endif
 
-# enable internal diagnostic logging
+# enable internal diagnostic logging?
 ifeq ($(SIR_SELFLOG),1)
-	CFLAGS += -DSIR_SELFLOG
-endif
-
-ifeq ($(SIR_USE_HASH),1)
-	CFLAGS += -DSIR_USE_HASH
+  CFLAGS += -DSIR_SELFLOG
 endif
 
+# enable assertions?
 ifeq ($(SIR_ASSERT_ENABLED),1)
-	CFLAGS += -DSIR_ASSERT_ENABLED
+  CFLAGS += -DSIR_ASSERT_ENABLED
 endif
 
-# on Windows, automatically defined by the preprocessor.
+# disable system loggers?
 ifeq ($(SIR_NO_SYSTEM_LOGGERS),1)
-	CFLAGS += -DSIR_NO_SYSTEM_LOGGERS
+  CFLAGS += -DSIR_NO_SYSTEM_LOGGERS
```

Comment:
Ah, the intention was to use two spaces, and left the recipe prefix as the default tab, but it looks funny sometimes.  I'll try to consistently use spaces and change the prefix.  So we'll see if that looks nicer.  I think it will.

Reply:
Also, I really do have half a mind to run clang-format again and put { on the next line for functions, but only functions.